launchdarkly/client.go

```diff
@@ -0,0 +1,79 @@
+package launchdarkly
+
+import (
+	"net/http"
+	"encoding/json"
+	"bytes"
+	"io/ioutil"
+	"errors"
+	"strconv"
+)
+
+type Client struct {
+	AccessToken string
+}
+
+func (c *Client) GetStatus(path string) (int, error) {
+	expectedStatus := make(map[int]bool)
+	return c.execute("GET", path, nil, expectedStatus, nil)
+}
+
+func (c *Client) Get(path string, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("GET", path, nil, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Post(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("POST", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Patch(path string, body interface{}, expectedStatus map[int]bool, response interface{}) error {
+	_, err := c.execute("PATCH", path, body, expectedStatus, response)
+	return err
+}
+
+func (c *Client) Delete(path string, expectedStatus map[int]bool) error {
+	_, err := c.execute("DELETE", path, nil, expectedStatus, nil)
+	return err
+}
+
+func (c *Client) execute(method string, path string, body interface{}, expectedStatus map[int]bool, response interface{}) (int, error) {
+	requestBody, err := json.Marshal(body)
+	if err != nil {
+		return 0, err
+	}
+
+	req, err := http.NewRequest(method, "https://app.launchdarkly.com/api/v2"+path, bytes.NewBuffer(requestBody))
+	if err != nil {
+		return 0, err
+	}
+
+	req.Header.Set("Authorization", c.AccessToken)
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return 0, err
+	}
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return resp.StatusCode, err
+	}
+
+	println(method + " " + path + " returned HTTP status " + strconv.Itoa(resp.StatusCode))
+
+	if len(expectedStatus) > 0 && !expectedStatus[resp.StatusCode] {
```

Comment:
If the resp.StatusCode is not in the map, it is going to crash. Expected? If so, what is the point of the map, it could be a list or an optional list with ... as the last parameter.

Reply:
nvm, not going to crash. Only the `ok` value is going to be false. It should still be a list since the map value is always `true`

Reply:
I changed it to a list, but I initially used a map because I was looking for a Set, and internet tells me that this isn't needed in Go because you can emulate it with a map yadda yadda... 😁 

To be honest I'm not a fan of having to manually code a loop to look for an element in an array.

![image](https://user-images.githubusercontent.com/1826047/43124978-0631751a-8f29-11e8-9c8e-3da7bcbb24ea.png)